File: src/main/java/com/thebluehats/server/core/modules/HitCounterModule.java
package com.thebluehats.server.core.modules;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.thebluehats.server.game.managers.enchants.HitCounter;

import com.thebluehats.server.game.managers.enchants.Timer;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.UUID;

public class HitCounterModule extends AbstractModule {
    @Provides
    static HitCounter provideHitCounter(Timer<UUID> timer) {
        return new HitCounter(timer);
    }

    @Override
    protected void configure() {
    }
}

File: src/main/kotlin/com/thebluehats/server/core/modules/HitCounterModule.kt
package com.thebluehats.server.core.modules

import com.google.inject.AbstractModule
import com.google.inject.Provides
import com.thebluehats.server.game.managers.enchants.HitCounter
import com.thebluehats.server.game.managers.enchants.Timer
import java.util.*

class HitCounterModule : AbstractModule() {
    override fun configure() {}

    companion object {
        @Provides
        @JvmStatic
        fun provideHitCounter(timer: Timer<UUID>): HitCounter {
            return HitCounter(timer)
        }
    }
}